internal/jobtokens/jobtokens.go

```diff
@@ -47,14 +47,14 @@ func New(n int) (*JobTokens, error) {
 
 // Take is how a job (goroutine) can Take its turn.
 func (jt *JobTokens) Take() {
-	jt.wg.Add(1)
 	<-jt.tokens
+	jt.wg.Add(1)
```

Comment:
To be able to wait for all ongoing jobs, including those waiting on getting their turn in `jt.Take`, we need to call `jt.wg.Add` before acquiring the token. This allows the internal waitgroup to account for a job waiting to acquire a token.